packages/react/collection/src/Collection.tsx

```diff
@@ -96,18 +96,19 @@ function createCollection<ItemElement extends HTMLElement, ItemData>() {
 
   function useCollection() {
     const context = React.useContext(Context);
-    return {
-      getItems() {
-        const orderedNodes = Array.from(
-          context.collectionRef.current!.querySelectorAll(`[${ITEM_DATA_ATTR}]`)
-        );
-        const items = Array.from(context.itemMap.values());
-        const orderedItems = items.sort(
-          (a, b) => orderedNodes.indexOf(a.ref.current!) - orderedNodes.indexOf(b.ref.current!)
-        );
-        return orderedItems;
-      },
-    };
+
+    const getItems = React.useCallback(() => {
+      const collectionNode = context.collectionRef.current;
+      if (!collectionNode) return [];
+      const orderedNodes = Array.from(collectionNode.querySelectorAll(`[${ITEM_DATA_ATTR}]`));
```

Comment:
Ah nice, I made the same change [here](https://github.com/radix-ui/primitives/pull/852/files#diff-0ad5c595db850ab96b0d987a0bd230537711c920e37ce92ac64e6d2876dd429cR106) but it hasn't been merged into `main` yet. Great minds.

Reply:
Brilliant 🙂